doc/tutorials/gapi/face_beautification/face_beautification.markdown

```diff
@@ -0,0 +1,388 @@
+object:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp net_param
+
+Every `cv::gapi::ie::Params<>` object is related to the network specified by its
+template argument. We should pass there the network type we have described in
+the previous step.
+Those objects are able to contain paths to structure and weights files and
+a desirable backend which can be one of the backends supported by
+InferenceEngine ("CPU", "GPU", etc.).
+
+All the nets' parameter objects should be contained in an object returned by
+`cv::gapi::networks()` function:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp netw
+
+To allow a certain pipeline to use our custom kernels (as well as all other
+non-standard kernels, e.g. `cv::gapi::core::fluid::kernels()`), we have to
+describe them by `cv::gapi::kernels<>` object:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp kern_pass_1
+
+After that, we must pass networks' parameters and kernels' descriptions to the
+graph by compile arguments:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp apply
+
+# Video stream processing with G-API {#gapi_fb_streaming}
+
+There is no interest in using the most of computer vision pipelines to process
+just a single picture --- a big group of them are created to be launched with a
+videosource as an input. Therefore, it is essential for G-API as an
+interface for pipelines implementation to provide a possibility of a stream
+processing without any external instruments to be applied by user.
+So, that's why the G-API feature providing such a possibility is needed.
+
+## Videostream processing before OpenCV 4.2 release {#gapi_fb_str_before}
+
+A videostream used to be captured by `cv::VideoCapture` object:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp bef_cap
+
+and then a compiled graph used to be applied to every single frame:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp bef_loop
+
+## Videostream processing after OpenCV 4.2 release {#gapi_fb_str_after}
+
+Now this can be done without explicit use of additional libraries. First of all,
+let's define a stream variable and pass to it graph compile arguments (which
+the graph will use within the stream):
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp str_comp
+
+Then we should set a source:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp str_src
+
+And for this moment our stream is ready to be launched! An example of a
+Streaming loop:
+
+@snippet cpp/tutorial_code/gapi/face_beautification/face_beautification.cpp str_loop
+
+There are some other useful possibilities provided, e.g. you can abort stream
+processing via `cv::GStreamingCompiled::stop()` method (the next call of
+`cv::GStreamingCompiled::start()` will start processing the stream from the very
+beginning).
```

Comment:
Should I leave the link to the GStreamingCompiled page in documentation in this place?

Reply:
It resolves automatically